lib/spack/spack/stage.py

```diff
@@ -241,10 +242,7 @@ class Stage:
     similar, and are intended to persist for only one run of spack.
     """
 
-    """Shared dict of all stage locks."""
-    stage_locks: Dict[str, spack.util.lock.Lock] = {}
```

Comment:
Also this I think might possibly lead to races. Also, I don't see why we would need that, since locks are essentialy synchronized at the filesystem level

Reply:
file system locks are process safe but not thread safe